packages/core/core/src/services/document-service/transform/id-transform.ts

```diff
@@ -34,14 +34,25 @@ async function transformParamsDocumentId(
     data = await transformRelationDataIds(idMap, input.data, { ...opts, uid });
   }
 
-  // TODO: Transform fields
-  const fields = input.fields;
-  // TODO: Transform filters
-  const filters = input.filters;
-  // TODO: Transform populate
-  const populate = input.populate;
-  // TODO: Transform sort
-  const sort = input.sort;
+  let filters = input.filters;
+  if (input.filters) {
+    filters = await transformFiltersOrPopulate(idMap, input.filters, { ...opts, uid });
+  }
+
+  // let populate = input.populate;
+  // if (input.populate) {
+  //   populate = await transformFiltersOrPopulate(idMap, input.populate, { ...opts, uid });
+  // }
+
+  let fields = input.fields;
+  if (input.fields) {
+    fields = transformFieldsOrSort(input.fields);
+  }
+
+  let sort = input.sort;
+  if (input.sort) {
+    sort = transformFieldsOrSort(input.sort);
+  }
```

Comment:
Even if dupped, I would keep two methods for fields and sort, we might find use cases where they have differences